### PR TITLE
cpu: Add setvcpus cases

### PR DESCRIPTION
**Pull request by Yingshun**
This PR adds setvcpus related cases.They are as follows:
1. Update "maximum/current" vcpus numbers by setvcpus
2. Hot-plug/unplug vCPU without <acpi> feature
3. Restart VM after changing inactive maxvcpu/current vcpus conf
4. Check maximum vcpus number with topology settings when guest is
shutdown/running
Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result:**
```
(.libvirt-ci-venv-ci-runtest-VcvqjV) [root@yicui-host1 domain]# avocado run --vt-type libvirt --vt-machine-type q35 virsh.setvcpus.error_test.no_acpi virsh.setvcpus.normal_test.guest_on.name_option.default  --vt-connect-uri qemu:///system
JOB ID     : 445868c87adec937e638c5ab240f06cfa9b54ec7
JOB LOG    : /root/avocado/job-results/job-2019-12-22T22.24-445868c/job.log
 (1/4) type_specific.io-github-autotest-libvirt.virsh.setvcpus.error_test.no_acpi: PASS (162.66 s)
 (2/4) type_specific.io-github-autotest-libvirt.virsh.setvcpus.normal_test.guest_on.name_option.default.option_maxmum_config: PASS (21.68 s)
 (3/4) type_specific.io-github-autotest-libvirt.virsh.setvcpus.normal_test.guest_on.name_option.default.restart_vm.update_cur: PASS (42.08 s)
 (4/4) type_specific.io-github-autotest-libvirt.virsh.setvcpus.normal_test.guest_on.name_option.default.restart_vm.update_max: PASS (42.22 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 271.28 s
(.libvirt-ci-venv-ci-runtest-VcvqjV) [root@yicui-host1 domain]# 
```

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg [hunk 1]
```diff
@@ -98,3 +114,9 @@
                             setvcpus_options = "--maximum --config"
                 - maximum_option:
                     setvcpus_options = "--maximum --live"
+                - no_acpi:
+                    no_acpi = "yes"
+                    setvcpus_count = ${setvcpus_current}
+                    hotunplugin = "yes"
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg]: hotunplug

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg [hunk 2]
```diff
@@ -19,6 +19,22 @@
...
+                                    setvcpus_count = "max_vcpu + 8"
+                                - restart_vm:
+                                    restart_vm = "yes"
+                                    variants:
+                                        - update_cur:
+                                            setvcpus_options = "--config"
+                                            setvcpus_count = "2"
+                                        - update_max:
+                                            setvcpus_options = "--maximum --config"
+                                            setvcpus_count = "1"
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg]: why max is 1 ?

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg [hunk 3]
```diff
@@ -19,6 +19,22 @@
                             # less than setvcpus_max
                             setvcpus_count = "2"
                         -del:
+                        - @default:
+                            only name_option
+                            variants:
+                                - option_maxmum_config:
+                                    update_maxmum_config = "yes"
+                                    setvcpus_options = "--config"
+                                    setvcpus_count = "max_vcpu + 8"
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg]: max_vcpu is hard string, how "max_vcpu + 8"   can express vcpu count

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg (hunk 1 above)
Comment (Yingshun) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg]: updated.

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg (hunk 3 above)
Comment (Yingshun) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg]: by using eval().

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg (hunk 2 above)
Comment (Yingshun) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_setvcpus.cfg]: the value is copied from manual case.

**Comment by Yingshun** [comment]
Add one more test to check maximum vcpus with topology settings when guest is
shutdown/running:

**Test results:**
```
(.libvirt-ci-venv-ci-runtest-VcvqjV) [root@yicui-host1 domain]# avocado run --vt-type libvirt --vt-machine-type q35 virsh.setvcpus.error_test.no_acpi virsh.setvcpus.normal_test.guest_on.name_option.default virsh.setvcpus.error_test.shut_off_error_option.with_topology virsh.setvcpus.error_test.maximum_option_with_topology --vt-connect-uri qemu:///system
JOB ID     : e7138d12545736c25b11e227a9e055ce92596c4c
JOB LOG    : /root/avocado/job-results/job-2019-12-27T03.40-e7138d1/job.log
 (1/6) type_specific.io-github-autotest-libvirt.virsh.setvcpus.error_test.no_acpi: PASS (164.26 s)
 (2/6) type_specific.io-github-autotest-libvirt.virsh.setvcpus.normal_test.guest_on.name_option.default.option_maxmum_config: PASS (22.15 s)
 (3/6) type_specific.io-github-autotest-libvirt.virsh.setvcpus.normal_test.guest_on.name_option.default.restart_vm.update_cur: PASS (43.19 s)
 (4/6) type_specific.io-github-autotest-libvirt.virsh.setvcpus.normal_test.guest_on.name_option.default.restart_vm.update_max: PASS (42.92 s)
 (5/6) type_specific.io-github-autotest-libvirt.virsh.setvcpus.error_test.shut_off_error_option.with_topology: PASS (24.99 s)
 (6/6) type_specific.io-github-autotest-libvirt.virsh.setvcpus.error_test.maximum_option_with_topology: PASS (22.33 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 324.39 s
(.libvirt-ci-venv-ci-runtest-VcvqjV) [root@yicui-host1 domain]# 
```

File: libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py [hunk 4]
```diff
@@ -191,9 +212,29 @@ def set_expected(vm, options):
...
+                vm.reboot()
+
             status = virsh.setvcpus(dom_option, count_option, options,
                                     ignore_status=True, debug=True)
             if not status_error:
+                if restart_vm:
+                    if vm.is_alive():
+                        vm.destroy()
+                    vm.start()
+                    vm.wait_for_login()
```
Comment (chloerh) [inline libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py]: If we do not want to use the session that 'wait_for_login()' created, it's better to close it after.

Comment (chloerh) [reply libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py]: `vm.wait_for_login().close()`

Comment (Yingshun) [reply libvirt/tests/src/virsh_cmd/domain/virsh_setvcpus.py]: updated.
